### ISSUE TITLE
Plugin installation

### ISSUE THREAD

**Issue by liaprins-czi** [napari/product-heuristics-2020#10]
Current website doesn't list example plugins to try, searching for plugins to play with don't surface anything easily.

Unclear how to install plugins, and whether installation worked. So far installation of attempted plugins has not worked.
__________________________________________________________
**Recommendation**

Known issue; Justin is leading plugin service incubation phase.

Provide better documentation about how to install plugins (not just how to create them).

Design better napari-specific confirmation + feedback about plugin installation success or errors (beyond Terminal errors if possible).

Critical issue
◆◆◆   High content writing work
●●○   Medium design work
▽▽▽   No development work

**Comment by goanpeca** [comment]
What is the status of this issues @napari/core-devs  :) ?

Any actions points we could start tackling on separate issues?

Thanks!

**Comment by sofroniewn** [comment]
@neuromusic is leading some efforts on the newly named `napari-hub` that will be the web destination for finding plugins. From there we actually want to enable a `one-click` installation that will open up the current napari plugin installer (which I think wasn't merged when this issue was made). Your thoughts/ advice on enabling such a one click install from the web might be appreciated. Curious @tlambert03 if you have ideas on it too.

I think the current thoughts are that you'd click install on the website and then you'd get one of those dialogs saying "this website is trying to open napari, do you allow it, then you'd get napari open with the plugin installer and if would have found the corresponding plugin inside the the plugin installer, high lighted it, and then you'd click "install" and we'd just do our conventional install procedure.

Very open to extra ideas here

**Comment by tlambert03** [comment]
> Your thoughts/ advice on enabling such a one click install from the web might be appreciated.

I looked into this a little bit (on mac at least).  We should be able to register a custom URL scheme ([CFBundleURLSchemes](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleurltypes/cfbundleurlschemes)) with the OS, which would let `napari://...` open up the napari application (presumably we could have some app args that could then do something like install a plugin).  There's a somewhat related example using applescript [here](http://jones.kaist.edu/notebook/2012/09/creating-a-custom-url-scheme-via-applescript-and-python.html).  However, I haven't yet been able to get it to work.

not sure how it's done on windows though

**Comment by sofroniewn** [comment]
Ok, sweet, not surprising that this may end up being platform specific. Thanks for the leads though! cc @justinelarsen

**Comment by neuromusic** [comment]
Nice! A couple of thoughts...

As @liaprins-czi noted, even before "one-click", there are improvements that can happen:

> Provide better documentation about how to install plugins (not just how to create them).
> 
> Design better napari-specific confirmation + feedback about plugin installation success or errors (beyond Terminal errors if possible).

I'll add a +1 to both of these. A guide to the "install/uninstall packages" dialog at https://napari.org/plugins/stable/index.html would be immediately useful and go a long way I think.

As for the napari hub, for our first iteration (June release) we only planning to provide clear, consistent descriptions on how to install each plugin. Once there's a path forward on how we might implement `napari://...` links across platforms, I'm confident that one-click install would be incredibly valuable to the community.

There are a bunch of details that would need to be worked through to ensure a smooth "one click" user experience, however:

Would this only work for the bundled app? What if a plugin appears in the hub, but not in the install dialog? Do we omit the one-click link for Python 3.7-only and 3.9-only plugins? Do we omit them from being listed in the hub? How do we handle informing users of dependencies that pip can't manage? How do we handle failures?

In other words, what are the edge cases in @sofroniewn 's proposed user journey?

None of these are unsolvable problems, but they are challenges that I think would benefit from some deliberate, coordinated focus from design+engineering when the time comes.

**Comment by liaprins-czi** [comment]
I have just created a proposal for small updates to napari.org that would help:
- plugin searchers understand how to install plugins within napari
- plugin searchers know about napari hub for finding plugins
- plugin developers know how to properly submit information about their plugin so it can be displayed on napari hub.

Here is the [G-Doc with the overall proposal](https://docs.google.com/document/d/1-O1nUl75jPxnH1v6xVewOetFpGPchtThVjI2cEWsvSs/edit?usp=sharing) + here is a [Figma with mockups](https://www.figma.com/file/NmdYwtpKfMzLnNLYYWrbTw/napari.org-final-designs?node-id=1362%3A0) of what the proposed changes would look like. Any feedback as comments in the G-Doc would be very helpful! Thanks!

(Just posting in this issue because it is related, and this proposal would help to solve this GH issue.)

**Comment by sofroniewn** [comment]
Love the figma mocks, all the new sections look very clear and easy to add. We can time this addition with the launch of the hub!
